Neos.Flow/Classes/Http/ContentStream.php

```diff
@@ -0,0 +1,314 @@
+<?php
+namespace Neos\Flow\Http;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Implementation of a PSR-7 HTTP stream
+ */
+class ContentStream implements StreamInterface
+{
+    /**
+     * @var resource
+     */
+    protected $resource;
+
+    /**
+     * @var string|resource
+     */
+    protected $stream;
+
+    /**
+     * @param string|resource $stream
+     * @param string $mode Mode with which to open stream
+     * @throws \InvalidArgumentException
+     */
+    public function __construct($stream, $mode = 'r')
+    {
+        $this->replace($stream, $mode);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function close()
+    {
+        if (!$this->resource) {
+            return;
+        }
+
+        $resource = $this->detach();
+        fclose($resource);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function detach()
+    {
+        $resource = $this->resource;
+        $this->resource = null;
+
+        return $resource;
+    }
+
+    /**
+     * Attach a new stream/resource to the instance.
+     *
+     * @param string|resource $stream
+     * @param string $mode
+     */
+    public function replace($stream, $mode = 'r')
+    {
+        $this->close();
+        $this->resource = $stream;
+        if (!is_resource($stream)) {
+            $this->openStream($stream, $mode);
+        }
+
+        if (!$this->hasValidResource()) {
+            throw new \InvalidArgumentException('Invalid stream provided; must be a string or stream resource', 1453891861);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSize()
+    {
+        if (!$this->hasValidResource()) {
+            return null;
```

Comment:
I somewhat agree but the Interface states throws for some methods, not for this one though. We could still throw one here but other implementations generally do not.